### PR TITLE
fix(manager/regex): set replaceString for digest-only matches

### PR DESCRIPTION
**Pull request by setchy**
## Changes

Sets `replaceString` when currentDigest match group is present. 

## Context

Resolves https://github.com/renovatebot/renovate/discussions/21340

Ran against https://github.com/setchy/renovate-bitbucket-pipeline-issue/pull/5/files (PR updated to expected values)

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/regex/strategies.ts [hunk 1]
```diff
@@ -44,7 +44,10 @@ export function handleCombination(
   const extraction = matches
     .map((match) => ({
       groups: match.groups ?? {},
-      replaceString: match?.groups?.currentValue ? match[0] : undefined,
+      replaceString:
+        match?.groups?.currentValue || match?.groups?.currentDigest
```
Comment (viceice) [inline lib/modules/manager/regex/strategies.ts:48]:
```suggestion
        match?.groups?.currentValue ?? match?.groups?.currentDigest
```
what happens when version and digest are in separate regex matches? would that currently also fail?

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/modules/manager/regex/strategies.ts (hunk 1 above)
Comment (setchy) [reply lib/modules/manager/regex/strategies.ts]: I believe it's OK due to the `.reduce` function on line 52.

**Review by viceice — APPROVED** [review]
LGTM, but requesting more reviews to be safe

**Comment by secustor** [comment]
Is this isolated to the `combination` strategy? 

I suspect we have the same issue with `any` and `recursive`.

**Comment by setchy** [comment]
> Is this isolated to the `combination` strategy? 
> 
> I suspect we have the same issue with `any` and `recursive`.

I did look at those and if I recall correctly, they always will set `replaceString = match[0]`

**Review by secustor — APPROVED** [review]

**Pull request merged**